File: 201-300/0232-My-Queue/0232.py
class MyQueue(object):

  def __init__(self):
    self.inStack = []
    self.outStack = []

  def push(self, x):
    self.inStack.append(x)

  def pop(self):
    if not self.outStack:
      while self.inStack:
        self.outStack.append(self.inStack.pop())
    return self.outStack.pop()

  def peek(self):
    if not self.outStack:
      while self.inStack:
        self.outStack.append(self.inStack.pop())
    return self.outStack[-1]

  def empty(self):
    return not self.inStack and not self.outStack

File: 201-300/0232-My-Queue/0232.java
class MyQueue {
  Deque<Integer> inStack;
  Deque<Integer> outStack;

  public MyQueue() {
    inStack = new LinkedList<Integer>();
    outStack = new LinkedList<Integer>();
  }

  public void push(int x) {
    inStack.push(x);
  }

  public int pop() {
    if (outStack.isEmpty()) {
      in2Out();
    }
    return outStack.pop();
  }

  public int peek() {
    if (outStack.isEmpty()) {
      in2Out();
    }
    return outStack.peek();
  }

  public boolean empty() {
    return inStack.isEmpty() && outStack.isEmpty();
  }

  private void in2Out() {
    while(!inStack.isEmpty()) {
      outStack.push(inStack.pop());
    }
  }
}